src/workspacesProvider.ts

```diff
@@ -57,7 +57,9 @@ export class WorkspaceProvider implements vscode.TreeDataProvider<vscode.TreeIte
     if (element) {
       if (element instanceof WorkspaceTreeItem) {
         const agents = extractAgents(element.workspace)
-        const agentTreeItems = agents.map((agent) => new AgentTreeItem(agent, element.watchMetadata))
+        const agentTreeItems = agents.map(
+          (agent) => new AgentTreeItem(agent, element.workspaceOwner, element.workspaceName, element.watchMetadata),
```

Comment:
Why not just pass `element` directly into the `AgentTreeItem`?

Reply:
Ideally, you could instantiate an agentTreeItem without a WorkspaceTreeItem, but it doesn't matter in practice. Thank you for the suggestion I will squeeze this in later :)